File: backend/Terena.Services/Database/Court.cs
using System.Collections.Generic;

namespace Terena.Services.Database
{
    public class Court
    {
        public int Id { get; set; }
        public int VenueId { get; set; }
        public string CourtType { get; set; }
        public string Name { get; set; }
        public bool IsAvailable { get; set; }
        public string MaxCapacity { get; set; }
        public Venue Venue { get; set; }
        public ICollection<Booking> Bookings { get; set; }
    }
}

File: frontend/terena_desktop/lib/models/court.dart
class Court {
  int? id;
  int? venueId;
  String? venueName;
  String? courtType;
  String? name;
  bool? isAvailable;
  String? maxCapacity;

  Court({
    this.id,
    this.venueId,
    this.venueName,
    this.courtType,
    this.name,
    this.isAvailable,
    this.maxCapacity,
  });

  Court.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    venueId = json['venueId'];
    venueName = json['venueName'];
    courtType = json['courtType'];
    name = json['name'];
    isAvailable = json['isAvailable'];
    maxCapacity = json['maxCapacity'];
  }

  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{};
    data['id'] = id;
    data['venueId'] = venueId;
    data['venueName'] = venueName;
    data['courtType'] = courtType;
    data['name'] = name;
    data['isAvailable'] = isAvailable;
    data['maxCapacity'] = maxCapacity;
    return data;
  }
}

